src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -0,0 +1,66 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.InvalidationListener;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a number with a {@link Spinner}
+ */
+public class NumberSpinnerInputSocketView extends InputSocketView<Number> {
+
+    private final static Object[] DEFAULT_DOMAIN = new Object[]{Double.MIN_VALUE, Double.MAX_VALUE};
+
+    private final SpinnerValueFactory<Double> valueFactory;
+    private final InvalidationListener updateSocketFromSpinner;
+
+    /**
+     * @param socket An <code>InputSocket</code> with a domain containing two <code>Number</code>s (the min and max
+     *               slider values), or no domain at all.
+     */
+    public NumberSpinnerInputSocketView(EventBus eventBus, InputSocket<Number> socket) {
+        super(eventBus, socket);
+
+
+        Object[] domain = socket.getSocketHint().getDomain();
+        if (domain == null) {
+            domain = DEFAULT_DOMAIN;
+        }
+
+        checkArgument(domain.length == 2 && domain[0] instanceof Number && domain[1] instanceof Number,
+                "Spinners must have a domain with two numbers (min and max)");
```

Comment:
One way or another, this switch statement needs to exist in the gui package. There is absolutely no way it can be anywhere else because it needs to create JavaFX controls.   

And the default case and the else statements that are invalid need to do something, and throwing an exception seems reasonable. (now that I think about it, using an assert might be better, but I won't try to justify that right now) 

Since these conditionals that throw exceptions already do the validation in this class, there's no need to do the same validation in another class. It would just be copying and pasting this same switch statement into SocketHint. 

The reason why the validation is done here is because the factory is the class capable of making sockethints to views, so it should be the one to validate if a certain mapping is possible. A socket is not invalid because we can't make a JavaFX control for it, and like you said, a class should validate itself. 

Although this validation was originally supposed to be in Inputsocketview, this was really the intention with socket hints from the beginning, not the hacked-together solution you seem to think.